GVFS/GVFS.Common/Tracing/EventListener.cs

```diff
@@ -8,36 +8,41 @@ public abstract class EventListener : IDisposable
         private readonly EventLevel maxVerbosity;
         private readonly Keywords keywordFilter;
 
+        private IEventListenerEventSink eventSink;
+
         protected EventListener(EventLevel maxVerbosity, Keywords keywordFilter)
         {
             this.maxVerbosity = maxVerbosity;
             this.keywordFilter = keywordFilter;
         }
 
+        public void RegisterEventSink(IEventListenerEventSink eventSink)
+        {
+            if (this.eventSink != null)
+            {
+                throw new Exception($"Already registered an event sink for event listener {this.GetType().Name}");

```

Comment:
Would `InvalidOperationException` be more appropriate? Not that it really matters...